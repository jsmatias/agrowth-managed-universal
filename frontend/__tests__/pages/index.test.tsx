import { withRouter } from 'next/router';
import React from 'react';
// tslint:disable-next-line:no-implicit-dependencies
import { render } from 'react-testing-library';
// tslint:disable-next-line:import-name
import HomePage from '../../src/pages/index';

// tslint:disable-next-line:no-console
console.error = jest.fn();

jest.mock('next/router', () => ({
  withRouter: jest.fn(() => () => (
    <h1 data-testid='withRouter'>Mocked withRouter</h1>
  ))
}));

describe('<HomePage/>', () => {
  test('should render', () => {
    const { queryByText, queryByTestId } = render(<HomePage />);
    expect(queryByText('Hello world')).toBeTruthy();
    expect(queryByTestId('withRouter')).toBeTruthy();
  });

  test('should call withRouter', () => {
    expect(withRouter).toHaveBeenCalledTimes(1);
  });

  test('should not raise errors', () => {
    // tslint:disable-next-line:no-console
    expect(console.error).not.toBeCalled();
  });

  test('should match snapshot', () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });
});
