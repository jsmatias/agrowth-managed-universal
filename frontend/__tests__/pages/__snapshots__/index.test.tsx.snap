// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage/> should match snapshot 1`] = `
<div>
  <h1
    data-testid="withRouter"
  >
    Mocked withRouter
  </h1>
  <main>
    <h1>
      Hello world
    </h1>
  </main>
</div>
`;
